fluent-bit-kinesis.go

```diff
@@ -160,7 +155,27 @@ func newKinesisOutput(ctx unsafe.Pointer, pluginID int) (*kinesis.OutputPlugin,
 		return nil, fmt.Errorf("[kinesis %d] Invalid 'compression' value (%s) specified, must be 'zlib', 'gzip', 'none', or undefined", pluginID, compression)
 	}
 
-	return kinesis.NewOutputPlugin(region, stream, dataKeys, partitionKey, roleARN, kinesisEndpoint, stsEndpoint, timeKey, timeKeyFmt, logKey, replaceDots, concurrencyInt, concurrencyRetriesInt, isAggregate, appendNL, comp, pluginID)
+	var httpRequestTimeoutDuration time.Duration
+	if httpRequestTimeout != "" {
+		httpRequestTimeoutInt, err := parseNonNegativeConfig("http_request_timeout", httpRequestTimeout, pluginID)
+		if err != nil {
+			return nil, err
+		}
+		httpRequestTimeoutDuration = time.Duration(httpRequestTimeoutInt) * time.Second
+	}
+
+	return kinesis.NewOutputPlugin(region, stream, dataKeys, partitionKey, roleARN, kinesisEndpoint, stsEndpoint, timeKey, timeKeyFmt, logKey, replaceDots, concurrencyInt, concurrencyRetriesInt, isAggregate, appendNL, comp, pluginID, httpRequestTimeoutDuration)
+}
+
+func parseNonNegativeConfig(configName string, configValue string, pluginID int) (int, error) {
+	configValueInt, err := strconv.Atoi(configValue)
+	if err != nil {
+		return 0, fmt.Errorf("[kinesis %d] Invalid '%s' value (%s) specified: %v", pluginID, configName, configValue, err)
+	}
+	if configValueInt < 0 {
```

Comment:
it depends on the individual configs and their subsequent usage. this was just a refactor of some existing repetitive code based on feedback.

side note: looks like there is an existing bug where setting the `experimental_concurrency_retries` to `0` when using `experimental_concurrency > 0` will swallow records without actually sending to kinesis since this for-loop condition will never be satisfied and the `Flush` call is inside it: https://github.com/aws/amazon-kinesis-streams-for-fluent-bit/blob/301b29e66f3fba6da18de9b662e52ed97add5058/kinesis/kinesis.go#L350

Reply:
So the fix here is to change the condition to `<=`? @zhonghui12

Reply:
> So the fix here is to change the condition to `<=`? @zhonghui12

ya that seems like it would be the simplest fix